### PR TITLE
Migrate download-artifacts Action

### LINKED ISSUE #3: 📋 [TASK] - Migrate download-artifacts Action
### Parent Feature

#1

### Task Description

As an AaC CI/CD Developer, I want to migrate the download-artifacts version, so that I have the latest supported pipeline action version.

### Acceptance Criteria

- [ ] Migrate versions in python-build-and-lint.yml template workflow
- [x] Migrate versions in python-cross-platform-support.yml template workflow
- [x] Migrate versions in python-deploy-artifacts.yml template workflow
- [ ] Migrate versions in python-test-python-version.yml template workflow
- [x] Include dynamic name referencing in download steps
  - [x] Repo and package version

**Helpful Links**

[Upload & Download @v4 action breakage](https://github.com/actions/upload-artifact/issues/472)

[download-artifact/docs/MIGRATION.md at main · actions/download-artifact (github.com)](https://github.com/actions/download-artifact/blob/main/docs/MIGRATION.md)

[download-artifact/README.md at main · actions/download-artifact (github.com)](https://github.com/actions/download-artifact/blob/main/README.md)

**Examples**

[Bump actions/download-artifact from 3 to 4 (#1186) · pypa/hatch@03b1bca (github.com)](https://github.com/pypa/hatch/commit/03b1bca8dcf0cbe6d43d9c014d5df47bcde036d6#diff-c4330c2cfbadba3e17aef4105738e5644a48e08e86096a32e2a6625fcd83b957)

[feat: bump actions/{download,upload}-artifact by samypr100 · Pull Request #1947 · astral-sh/uv (github.com)](https://github.com/astral-sh/uv/pull/1947/files)

[chore(deps): bump actions/download-artifact from 3 to 4 by dependabot[bot] · Pull Request #190 · appunite/Loudius (github.com)](https://github.com/appunite/Loudius/pull/190/files)

### Unresolved questions

_No response_

### PR DESCRIPTION
**Pull request by bunchw**
# Description

Bumping the version of the download-artifacts action, and adding dynamic repo referencing. 

# Linked Items:

Closes/Fixes/Resolves #3

### Added

Version migration for download-artifacts actions, and for one upload artifacts action that was missed in the previous PR.

### Changed

- _Describe any changes in existing functionality._

### Deprecated

- _Describe any deprecated features._

### Removed

- _Describe any removed features._

### Fixed

- _Describe any bug fixes._

### Security

- _Describe any security-related changes._

# Checklist:

- [ ] I updated project documentation to reflect my changes.
- [x] My changes generate no new warnings.
- [ ] I updated new and existing unit tests to account for my changes.
- [x] I linked the associated item(s) to be closed.
- [ ] I bumped the version.
- [x] I added the labels corresponding to my changes.

### CODE REVIEW

File: workflow-templates/issues2md.yml
Comment (lizzcondrey) [inline workflow-templates/issues2md.yml:1]: We don't need to worry about this workflow. And actually, can be removed. It's a legacy one that doesn't get used ever anymore.

File: workflow-templates/python-cross-platform-support.yml [hunk 1]
```diff
@@ -77,9 +77,9 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Download Secure Install
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         with:
-          name: secure_install_${{ matrix.os }}
+          name: secure_install_${{ matrix.os }}_${{steps.package-version.outputs.VERSION_STRING}}
```
Comment (lizzcondrey) [inline workflow-templates/python-cross-platform-support.yml:88]: This is in a separate job, so either need to add the additional step of generating the version string. Or, and really preferred, make a stand alone job of getting the version string that can be referenced in other jobs and their steps.

File: workflow-templates/python-deploy-artifacts.yml [hunk 2]
```diff
@@ -23,9 +23,9 @@ jobs:
     needs: build_deployment_artifacts
     steps:
       - name: Download Python Wheel
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         with:
-          name: ${{ github.event.repository.name }}_wheel
+          name: ${{ github.event.repository.name }}_${{steps.package-version.outputs.VERSION_STRING}}_wheel
```
Comment (lizzcondrey) [inline workflow-templates/python-deploy-artifacts.yml:34]: Need the step for creating the version string to be referenced.

**Review by lizzcondrey — CHANGES_REQUESTED** [review]
You're missing the build and lint workflow and the python test workflow according to the task list.

**Comment by bunchw** [comment]
No download-artifact steps exist in the build-and-lint or python-test-python-version workflows.

**Review by lizzcondrey — APPROVED** [review]
Should be good to go.

**Pull request merged**